src/operators/tests/join_test.cc

```diff
@@ -35,7 +35,7 @@ using namespace hustle::operators;
 
```

Comment:
not sure from a quick pass if there is a test that tests the join order change application
do you test that whatever way you pass in the relation to the join operator the hashtable is build on the smaller relation?

Reply:
Join order change application? 

(Actually, lot more change need to be done in the join space, which I realized after fixing this. I am planning to handle them in the resolver)

Reply:
Yes, previously, the left-hand side variable always considered as the larger table. (It resulted in correctness issue)

Previously, this query will be giving us the wrong answer (where the LHS is a small relation), after the fix it works correctly. 

 std::string query =
      "select sum(lo_extendedprice*lo_discount) as "
      "revenue "
      "from lineorder, ddate "
      "where **d_datekey = lo_orderdate** and d_year = 1993 and (lo_discount "
      "BETWEEN 0 and 3 and lo_quantity < 25);";


Also, the unit tests written in join_test.cc also tests for this case.

Reply:
so you test by verifying the correctness of the results. Ideally, you should test the functionality as well (i.e. verify that the map is built on the side you expect). Not as important here, feel freee to merge